### ISSUE TITLE
Performance Drop with New SIFT Feature Extraction

### ISSUE THREAD

**Issue by TianyiZhang-arc** [colmap/colmap#2955]
Hi all,
I encountered an issue while using the latest version of COLMAP (related to the issue #2911, #2898). I noticed a significant drop in pose accuracy for GLOMAP and COLMAP when running on ETH3D, compared to the previous version. Below, I outline the results of my experiments on the ETH3D/courtyard:
In the older version of COLMAP, the feature matcher produced non-deterministic results across runs. In my experiment, the results were similar to those reported in the GLOMAP paper, with the following AUC (1/3/5) (here I list the results from GLOMAP paper):
GLOMAP: 87.8 / 95.9 / 97.6
COLMAP: 87.3 / 95.8 / 97.5
After reinstalling the latest version of COLMAP, I observed consistent results across runs. However, the performance dropped significantly, as shown in the following AUC (1/3/5/10) in my experiments:
GLOMAP: 40.14 / 65.04 / 76.62 / 87.96
COLMAP: 56.77 / 83.59 / 90.15 / 95.07
Could you please help to check this? Thanks a lot!

**Comment by ahojnnes** [comment]
Thanks for reporting the issue. To help narrow down the issue, could you please confirm that you are using the CUDA SIFT implementation for feature extraction?

**Comment by TianyiZhang-arc** [comment]
Hi Johannes, thanks for your quick reply and helpful suggestions for narrowing down the problem!
I retested commit 3adff01 (right before #2929) and the latest one on the eth3d/courtyard (courtyard_dslr_undistorted.7z) using the original image size and the downsampled x8 dataset, with the following code. 
[code.zip](https://github.com/user-attachments/files/17941777/code.zip)

Here are some observations and questions:

1. On original image set, there is basically no difference between each version, in this case I think it may not be caused by the SIFT features #2929. Could you please explain the difference in performance reported in GLOMAP's experiments compared to ours?
COLMAP:AUC [1/3/5/10/20/]:53.4272/81.3482/88.6110/94.2887/97.1444/
GLOMAP:AUC [1/3/5/10/20/]:53.6064/78.9376/85.2097/89.9733/92.3551/

2. On the downsampled image set, there is slight variance between runs of COLMAP, but significant variance between runs of GLOMAP. To determine whether this is caused by differences in the database or the optimization process, I reran the experiments several times using the same database, as shown below. Could you help explain the reason for this behavior?

Database1, three different runs of glomap mapper:
AUC [1/3/5/10/20/]:16.0013/24.5276/27.0501/30.1548/35.5258/
AUC [1/3/5/10/20/]:17.7671/25.3381/27.6196/32.1874/39.8492/
AUC [1/3/5/10/20/]:17.4677/26.7125/30.3109/36.9238/48.1008/

Database2, three different runs of glomap mapper:
AUC [1/3/5/10/20/]:5.9410/11.6898/13.3481/15.6873/17.9642/
AUC [1/3/5/10/20/]:6.4551/13.3885/15.6598/17.6312/19.0153/
AUC [1/3/5/10/20/]:7.2033/12.7992/14.5560/16.2349/17.4213/

Thanks a lot for the help!

**Comment by ahojnnes** [comment]
Do you provide the known intrinsics? Do you fix the intrinsics during reconstruction?

**Comment by TianyiZhang-arc** [comment]
Hi Johannes, I gave several runs according to your guidance, and I discovered that the issue was related to glomap rather than the feature extraction. I passed the GT intrinsics, but I should also set '--BundleAdjustment.optimize_intrinsics False'. In my test, not setting this parameter doesn’t significantly impact the performance on the whole dataset. However, since we need to split several images for rendering, leaving this parameter unset caused variance across different runs in the split dataset. 
Thanks again for your help!
